### PR TITLE
[FIX] Routes retaining linkable while navigating to child routes

### PR DESCRIPTION
**Pull request by abhilashlr**
Fix for #42

### CODE REVIEW

**Comment by abhilashlr** [comment]
@poteto can we have this merged?

**Comment by jevanlingen** [comment]
@abhilashlr: Good you look to the changes, I really like this fix!

**Comment by jcbvm** [comment]
@abhilashlr ping

**Comment by abhilashlr** [comment]
Hello there,

Sorry that this has become a complete change in code or rather the PRs weren't accepted before, that I lost track of this repo.

Will resolve them when I get some free time and have a PR sent.

**Comment by abhilashlr** [comment]
Travis build fails with google-chrome-stable. @poteto any idea on how to fix this?

**Comment by dguettler** [comment]
@abhilashlr I think the issue is Ubuntu 12, which is no longer supported http://askubuntu.com/questions/758571/google-chrome-stable-depends-on-libstdc6-4-8-0-however-version-of-libs
So the solution would be to switch to a more recent version of Ubuntu, options would be 14.04 LTS or 16.04 LTS

**Comment by dguettler** [comment]
Correction, 14.04 LTS seems to be the only option as 16.04 LTS is not supported yet on Travis and wont for some time according to https://github.com/travis-ci/travis-ci/issues/5821

**Comment by cibernox** [comment]
This addon now uses circle-ci, so travis config is not an issue anymore.

**Comment by cibernox** [comment]
This seems to be the same as #94 /cc @samdemaeyer

**Comment by dguettler** [comment]
@cibernox so merging or rebasing this PR onto development should fix the issue

File: addon/components/bread-crumbs.js [hunk 1]
```diff
@@ -9,9 +9,10 @@ const {
   getWithDefault,
   assert,
   typeOf,
-  setProperties,
   A: emberArray,
-  String: { classify }
+  String: { classify },
+  copy,
+  merge
```
Comment (poteto) [inline addon/components/bread-crumbs.js]: Let's use `assign` instead, but don't put it in this destructure:

```js
const assign = Ember.assign || Ember.merge;
```

File: addon/components/bread-crumbs.js [hunk 2]
```diff
@@ -93,7 +94,7 @@ export default Component.extend({
       if (typeOf(breadCrumb) === 'null') {
         return;
       } else {
-        setProperties(breadCrumb, {
+        breadCrumb = merge(copy(breadCrumb), {
```
Comment (poteto) [inline addon/components/bread-crumbs.js]: `assign` please

File: .travis.yml [hunk 3]
```diff
@@ -1,28 +1,49 @@
 ---
+sudo: true
```
Comment (poteto) [inline .travis.yml]: This whole file can be deleted. Please rebase against current `master`

**Comment by fsmanuel** [comment]
@abhilashlr can you rebase and incorporate @poteto comments?

**Comment by noxan** [comment]
Since I was fighting the bug solved by this pull request, I rebased this branch and tried to incorporate the changes requested, see https://github.com/poteto/ember-crumbly/compare/master...noxan:master

What would be the best way to get those changes merged?

**Pull request closed without merging**